contracts/wrapper/NameWrapper.sol

```diff
@@ -413,10 +413,56 @@ contract NameWrapper is
         (address owner, uint32 oldFuses, uint64 expiry) = getData(
             uint256(node)
         );
-        _setFuses(node, owner, ownerControlledFuses | oldFuses, expiry);
+        _setFuses(node, owner, ownerControlledFuses | oldFuses, expiry, expiry);
         return ownerControlledFuses;
     }
 
+    /**
+     * @notice Extends expiry for a name
+     * @param parentNode Parent namehash of the name e.g. vitalik.xyz would be namehash('xyz')
+     * @param labelhash Labelhash of the name, e.g. vitalik.xyz would be keccak256('vitalik')
+     * @param expiry When the name will expire in seconds since the Unix epoch
+     * @return New expiry
+     */
+
+    function extendExpiry(
+        bytes32 parentNode,
+        bytes32 labelhash,
+        uint64 expiry
+    ) public returns (uint64) {
+        bytes32 node = _makeNode(parentNode, labelhash);
+
+        // this flag is used later, when checking fuses
+        bool canModifyParentName = canModifyName(parentNode, msg.sender);
+        // only allow the owner of the name or owner of the parent name
+        if (!canModifyParentName && !canModifyName(node, msg.sender)) {
+            revert Unauthorised(node, msg.sender);
+        }
+
+        (address owner, uint32 fuses, uint64 oldExpiry) = getData(
+            uint256(node)
+        );
+        if (owner == address(0) || ens.owner(node) != address(this)) {
+            revert NameIsNotWrapped();
+        }
```

Comment:
Shouldn't we be doing this check elsewhere (or not at all)? We don't want to be doing it on every method separately.

Reply:
I believe it is being done in several other places too, this logic was just added recently for the audit mitigations: [https://github.com/ensdomains/ens-contracts/pull/170/files#diff-8f057ac6433f8d6754bd5e1e9852faaf2d182b21cad6f5869f9041169c729c77R508](https://github.com/ensdomains/ens-contracts/pull/170/files#diff-8f057ac6433f8d6754bd5e1e9852faaf2d182b21cad6f5869f9041169c729c77R508)